pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -254,7 +270,7 @@ public void removeAgedDeletedSegments(int retentionInDays) {
           int numFilesDeleted = 0;
           for (String targetFile : targetFiles) {
             URI targetURI = URIUtils.getUri(targetFile);
-            Date dateToDelete = DateTime.now().minusDays(retentionInDays).toDate();
+            Date dateToDelete = DateTime.now().minus(getRetentionPeriodFromFile(targetFile)).toDate();
```

Comment:
Not introduced in this PR, but can be simplified:
```suggestion
            long deletionTimeMs = System.currentTimeMillis() - getRetentionMsFromFile(targetFile);
```